L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryHexLayerBasedImp1.cc

```diff
@@ -0,0 +1,737 @@
+            << module << ". The trigger cell neighbor mapping should be modified. \n";
+    }
+    const auto& neighbors = neighbors_itr->second;
+    // create HGCalDetId of neighbors and check their validity
+    neighbor_detids.reserve(neighbors.size());
+    for(const auto& module_tc : neighbors)
+    {
+        HGCalDetId neighbor_det_id((ForwardSubdetector)trigger_cell_det_id.subdetId(), trigger_cell_det_id.zside(), trigger_cell_det_id.layer(), 1, module_tc.first, module_tc.second);
+        if(validTriggerCell(neighbor_det_id.rawId()))
+        {
+            neighbor_detids.emplace(neighbor_det_id.rawId());
+        }
+    }
+    return neighbor_detids;
+}
+
+
+GlobalPoint 
+HGCalTriggerGeometryHexLayerBasedImp1::
+getTriggerCellPosition(const unsigned trigger_cell_det_id) const
+{
+    unsigned subdet = HGCalDetId(trigger_cell_det_id).subdetId();
+    // Position: barycenter of the trigger cell.
+    Basic3DVector<float> triggerCellVector(0.,0.,0.);
+    const auto cell_ids = getCellsFromTriggerCell(trigger_cell_det_id);
+    // BH
+    if(subdet==ForwardSubdetector::HGCHEB)
+    {
+        for(const auto& cell : cell_ids)
+        {
+            HcalDetId cellDetId(cell);
+            triggerCellVector += bhGeometry().getPosition(cellDetId).basicVector();
+        }
+    }
+    // EE or FH
+    else
+    {
+        for(const auto& cell : cell_ids)
+        {
+            HGCalDetId cellDetId(cell);
+            triggerCellVector += (cellDetId.subdetId()==ForwardSubdetector::HGCEE ? eeGeometry().getPosition(cellDetId) :  fhGeometry().getPosition(cellDetId)).basicVector();
+        }
+    }
+    return GlobalPoint( triggerCellVector/cell_ids.size() );
+
+}
+
+GlobalPoint 
+HGCalTriggerGeometryHexLayerBasedImp1::
+getModulePosition(const unsigned module_det_id) const
+{
+    unsigned subdet = HGCalDetId(module_det_id).subdetId();
+    // Position: barycenter of the module.
+    Basic3DVector<float> moduleVector(0.,0.,0.);
+    const auto cell_ids = getCellsFromModule(module_det_id);
+    // BH
+    if(subdet==ForwardSubdetector::HGCHEB)
+    {
+        for(const auto& cell : cell_ids)
+        {
+            HcalDetId cellDetId(cell);
+            moduleVector += bhGeometry().getPosition(cellDetId).basicVector();
+        }
+    }
+    // EE or FH
+    else
+    {
+        for(const auto& cell : cell_ids)
+        {
+            HGCalDetId cellDetId(cell);
+            moduleVector += (cellDetId.subdetId()==ForwardSubdetector::HGCEE ? eeGeometry().getPosition(cellDetId) :  fhGeometry().getPosition(cellDetId)).basicVector();
+        }
+    }
+    return GlobalPoint( moduleVector/cell_ids.size() );
+}
+
+
+void 
+HGCalTriggerGeometryHexLayerBasedImp1::
+fillMaps()
+{
+    //
+    // read module mapping file
+    std::ifstream l1tModulesMappingStream(l1tModulesMapping_.fullPath());
+    if(!l1tModulesMappingStream.is_open()) edm::LogError("HGCalTriggerGeometry") << "Cannot open L1TModulesMapping file\n";
```

Comment:
Shouldn't this skip trying to fill the maps if the stream can't be opened? (for every instance in this function and the next function)